unified-runtime/test/conformance/usm/umfIntegration.cpp

```diff
@@ -0,0 +1,59 @@
+// Copyright (C) 2025 Intel Corporation
+// Part of the Unified-Runtime Project, under the Apache License v2.0 with LLVM
+// Exceptions. See LICENSE.TXT
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#include "helpers.h"
+#include "uur/utils.h"
+#include <uur/fixtures.h>
+#include <uur/known_failure.h>
+
+#include <umf.h>
+#include <umf/memory_pool.h>
+#include <umf/memory_provider.h>
+
+struct umfDeviceAllocTest : uur::urUSMAllocTest {
+  void SetUp() override {
+    UUR_RETURN_ON_FATAL_FAILURE(uur::urUSMAllocTest::SetUp());
+    ASSERT_SUCCESS(uur::GetDeviceUSMDeviceSupport(device, USMSupport));
+    if (!USMSupport) {
+      GTEST_SKIP() << "Device USM is not supported.";
+    }
+  }
+};
+
+// The 0 value parameters are not relevant for umfDeviceAllocTest tests, they
+// are used below in urUSMDeviceAllocAlignmentTest for allocation size and
```

Comment:
can this be reduced to just a BoolTest for UsePool? I don't see the other params getting used anywhere in this file

Reply:
The umfDeviceAllocTest I'm using here is basically just an alias for uur::urUSMAllocTest that is implemented in another file and supports those other parameters. I'm not sure if I could get rid of the other params without rewriting urUSMAllocTest